API/src/Modules/XpSystems/Momentum.XpSystems.Infrastructure/InitialData.cs

```diff
@@ -0,0 +1,90 @@
+﻿using Marten;
+using Marten.Schema;
+using Momentum.XpSystems.Core.Models;
+using Momentum.XpSystems.Core.Models.Cosmetic;
+using Momentum.XpSystems.Core.Models.Rank;
+
+namespace Momentum.XpSystems.Infrastructure
+{
+    public class InitialData : IInitialData
+    {
+        private readonly object _initialData;
+
+        public InitialData(object initialData)
+        {
+            _initialData = initialData;
+        }
+        public void Populate(IDocumentStore store)
+        {
+            using var session = store.LightweightSession();
+
+            session.Store(_initialData);
+
+            session.SaveChanges();
```

Comment:
Regarding the issue here, theres a couple steps to clean it up and reduce the issues: 

1. Inject the `IXpSystemsRepository` - that way we don't need to repeat insertion/query logic.
2. First call `.Get()` (will become .SingleOrDefault(), see below comment). If the output is not null, return.
3. Now we know there is no initialized xp system, then run the createorupdate code